### PR TITLE
Finished

### PR DESCRIPTION
**Pull request by webmarkyn**

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  arr
+end
+
+def bubble_sort_by(arr)
+  (0..arr.length - 1).each do |i|
+    (1..(arr.length - 1) - i).each do |j|
+      if block_given?
+        arr[j], arr[j - 1] = arr[j - 1], arr[j] if yield(arr[j - 1], arr[j]).positive?
+      elsif arr[j - 1] > arr[j]
+        arr[j], arr[j - 1] = arr[j - 1], arr[j]
```
Comment (samgaco) [inline main.rb:18]: - [ ] If a block is not given, this function should raise an error

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @webmarkyn and @geraldgsh :high_brightness: 

- [ ] Are you working by pairs? I can only see commits by @webmarkyn 
- [ ] The second function implemented should be able to raise an error when no block is passed

----------------------------------
Remember to :arrow_down: 
* **[For your next code review requests]**  Add a short summary of changes in Pull Request description
--------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: main.rb (hunk 1 above)
Comment (geraldgsh) [reply main.rb:18]: @samgaco I've revised the code to include an error message for missing block. I'll submit a separate code review

Comment (geraldgsh) [reply main.rb:18]: @samgaco 
Thank you for taking the time to review our code

😄

File: .idea/inspectionProfiles/Project_Default.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="Rubocop" enabled="false" level="WARNING" enabled_by_default="false" />
+  </profile>
+</component>
```
Comment (bolah2009) [inline .idea/inspectionProfiles/Project_Default.xml:6]: Files not related to the project should be added to `.gitignore`.
I would suggest deleting these files and adding `.idea/*` to `.gitignore`.

**Review by bolah2009 — APPROVED** [review]
Please check the comments below before merging
It's a :+1: from me.

#### :heavy_check_mark: You can merge to master.

-----

### Let's make this Pull Request look more professional.
  - [NEXT PROJECT]: Use meaningful commits messages. [Learn more here](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/) and Check [this awesome article](https://chris.beams.io/posts/git-commit/)
  - [REQUIRED] Use meaningful names for  Pull Requests.
  - [REQUIRED] Add a short summary of changes in the Pull Request description.

#### Rename PR
  rename PR
-----
#### Update PR description
  update-PR-description-html

File: .idea/inspectionProfiles/Project_Default.xml (hunk 2 above)
Comment (geraldgsh) [reply .idea/inspectionProfiles/Project_Default.xml:6]: Thanks @bolah2009!
We'll implement your suggestions in our next project

**Pull request merged**